src/frontend/js/widgets/Dashboard/utils/dashboardRoutes.tsx

```diff
@@ -0,0 +1,93 @@
+import { IntlShape, MessageDescriptor, useIntl } from 'react-intl';
+import type {
+  FormatXMLElementFn,
+  Options as IntlMessageFormatOptions,
+  PrimitiveType,
+} from 'intl-messageformat';
+import { Navigate, RouteObject } from 'react-router-dom';
+import { isEnumValue } from 'types/utils';
+import { DashboardLayoutRoute } from 'widgets/Dashboard/components/DashboardLayoutRoute';
+import {
+  LearnerDashboardPaths,
+  learnerDashboardRouteLabels,
+  learnerDashboardRoutePaths,
+} from 'widgets/Dashboard/utils/learnerRouteMessages';
+import { getLearnerDashboardRoutes } from 'widgets/Dashboard/utils/learnerRoutes';
+import { getTeacherDashboardRoutes } from 'widgets/Dashboard/utils/teacherRoutes';
+import {
+  TeacherDashboardPaths,
+  teacherDashboardRouteLabels,
+  teacherDashboardRoutePaths,
+} from './teacherRouteMessages';
+
+export interface DashboardRouteHandle {
+  crumbLabel?: MessageDescriptor;
+  renderLayout?: boolean;
+}
+
+export function getDashboardRoutes() {
+  const intl = useIntl();
+  const getRoutePath = getDashboardRoutePath(intl);
+  const routes: RouteObject[] = [
+    {
+      path: '/',
+      element: <DashboardLayoutRoute />,
+      children: [
+        {
+          index: true,
+          element: <Navigate to={getRoutePath(LearnerDashboardPaths.COURSES)} replace />,
+        },
+        ...getLearnerDashboardRoutes(),
+        {
+          path: getRoutePath(TeacherDashboardPaths.ROOT),
+          handle: {
+            renderLayout: true,
+          },
+          children: getTeacherDashboardRoutes(),
+        },
+      ],
+    },
+  ];
+  return routes;
+}
+
+/**
+ * Use `intl.formatMessage` to retrieve a path or label of a given route for the active locale.
+ *
+ * Currying function which takes first the type of attribute to translate (path or label)
+ * then takes the intl object retrieved from the IntlProvider.
+ * Finally, it takes a path and options to pass to `intl.formatMessage` method
+ */
+
+enum RouteAttributes {
+  PATH = 'path',
+  LABEL = 'label',
+}
+
+const getDashboardRouteAttribute =
+  (attribute: RouteAttributes) =>
+  (intl: IntlShape) =>
+  (
+    path: LearnerDashboardPaths | TeacherDashboardPaths,
+    ...options: [
+      values?: Record<string, PrimitiveType | FormatXMLElementFn<string, string>>,
+      opts?: IntlMessageFormatOptions,
+    ]
+  ) => {
+    const isLabel = attribute === RouteAttributes.LABEL;
+
+    let message: MessageDescriptor;
+    if (isEnumValue(path, LearnerDashboardPaths)) {
+      message = isLabel ? learnerDashboardRouteLabels[path] : learnerDashboardRoutePaths[path];
```

Comment:
Maybe something like this could work ? 

```let message = learnerDashboardRouteLabels[path] ?? teacherDashboardRoutePaths[path]```

If find this `if .. else ..` logic quite complicated with the reversed ternary condition .. 🤔

Reply:
TS rale si on lui donne une clef impossible, ici `learnerDashboardRouteLabels[path]` sans le typeguard avant throw une erreur.